sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/ConfigurationManager.cs

```diff
@@ -2,77 +2,179 @@
 // Licensed under the MIT License.
 
 using System;
-using System.Collections.Generic;
 using System.Globalization;
-using System.Linq;
+using System.Net;
+using System.Net.Http.Headers;
+
+using Microsoft.IdentityModel.JsonWebTokens;
 
 namespace Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents
 {
+    /// <summary>
+    /// Configuration manager for loading up token validations.
+    /// </summary>
     internal class ConfigurationManager
     {
-        public static Dictionary<string, ServiceInfo> SERVICES = new Dictionary<string, ServiceInfo>()
-        {
-            { "99045fe1-7639-4a75-9d4a-577b6ca3810f", new ServiceInfo("https://login.microsoftonline.com","https://sts.windows.net/{0}/","https://login.microsoftonline.com/{0}/v2.0"){DefaultService=true } } //Public cloud
-        };
-
-        private const string BYPASS_VALIDATION = "AuthenticationEvents__BypassTokenValidation";
-        private const string CUSTOM_CALLER_APPID = "AuthenticationEvents__CustomCallerAppId";
-        internal const string TENANT_ID = "AuthenticationEvents__TenantId";
-        internal const string AUDIENCE_APPID = "AuthenticationEvents__AudienceAppId";
-        internal const string TOKEN_V1_VERIFY = "appid";
-        internal const string TOKEN_V2_VERIFY = "azp";
+        private const string OpenIdConfigurationPath = "/.well-known/openid-configuration";
+        private const string OpenIdConfigurationPathV2 = "/v2.0/.well-known/openid-configuration";
+
+        private const string AuthorityUrlKey = "AuthenticationEvents__AuthorityUrl";
```

Comment:
nit: I would set the AuthenticationEvents__ as prefix and then construct the keys